src/libexpr/flake/flake.cc

```diff
@@ -404,6 +418,12 @@ LockedFlake lockFlake(
                         // Respect the “flakeness” of the input even if we
                         // override it
                         i->second.isFlake = input2.isFlake;
```

Comment:
Note that this workaround could be replaced by a "proper" merge as now done with the other attributes below if `isFlake` had a value for "not specified", e.g. was changed to `std::optional<bool>`. But wanting to override the flakeness of an input does seem like an exotic use case.